### PR TITLE
Drinking glasses shatter when shot at

### PR DESCRIPTION
**Pull request by NightmareChamillian**
## About the PR 
Makes it so that drinking glasses can shatter and spill their contents if a projectile ``Cross``es over them

## Why's this needed? 
Quality of life, allows for shooting range gimmicks

## Changelog

```changelog
(u)NightmarechaMillian
(+)Drinking glasses now have a 33% chance to break when shot at - spectate a bar fight today!
```

### CODE REVIEW

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 1]
```diff
@@ -1301,6 +1301,13 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 	ex_act(severity)
 		src.smash()
 
+
+	Cross(atom/movable/mover) //epic barfights
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/tools/food_and_drink.dm:1305]:
```suggestion
	Crossed(atom/movable/mover) //epic barfights
```
`Cross` is to check if something is allowed to cross, `Crossed` is to do something when it does.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Very cool little detail, could probably be a higher chance.

**Comment by ThatFiveGuys** [comment]
Should probably be kinetic projectiles only and agree with it should being a higher chance, maybe scaling with bullet damage?

**Comment by BatElite** [comment]
30% seems pretty alright to me, in case you're aiming at someone behind the bar

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 2]
```diff
@@ -1301,6 +1301,13 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 	ex_act(severity)
 		src.smash()
 
+
+	Crossed(atom/movable/mover) //Makes barfights cooler
+		if(istype(mover, /obj/projectile))
+			if(prob(30))
+				src.smash()
+		..()
```
Comment (ZeWaka) [inline code/modules/chemistry/tools/food_and_drink.dm:1309]:
```suggestion
		. = ..()
```

**Review by ZeWaka — APPROVED** [review]

**Comment by Gannets** [comment]
I love anything that makes the environment react better to combat.

**Pull request merged**